zulip_bots/zulip_bots/bots/lunch/lunch.py

```diff
@@ -0,0 +1,214 @@
+            user = msg[0][1:].replace('**', '')
+            msg  = msg[1:]
+
+        # Find the command
+        if len(msg) == 0:
+            return  # null
+
+        # <user> I want that
+        if msg[-1] == 'that':
+            return copy_user_list(user, message['sender_full_name'])
+
+        # <user|optional> (I) want <item>
+        if msg[0:2] == ['I', 'want']:
+            return add_to_user_list(user, msg[2:])
+
+        if msg[0] == 'wants':
+            return add_to_user_list(user, msg[1:])
+
+        # <user|optional> remove all
+        if msg == ['remove', 'all']:
+            return remove_all(user)
+
+        # <user|optional> remove <item>
+        if msg[0] == 'remove':
+            return remove_from_user_list(user, msg[1:])
+
+        # orders
+        if msg[0] == 'orders':
+            return show_all_orders()
+
+        # <user|optional> order
+        if msg[0] == 'order':
+            return show_order(user)
+
+        # who should <verb>?
+        if msg[0:2] == ['who', 'should']:
+            return random_user(msg[2:])
+
+        if msg[0] == 'help':
+            return '''
+                <user|optional> (I) want(s) <item>
+                Adds <item> to the sender's (or <user>'s if specified) list of items
+
+                <user> I want that
+                Adds every item in <user>'s list of items to the sender's list of items
+
+                <user|optional> remove <item>
+                Removes <item> to the sender's (or <user>'s if specified) list of items
+
+                <user|optional> remove all
+                Removes all items to the sender's (or <user>'s if specified) list of items
+
+                orders
+                Show all orders
+
+                <user|optional> order
+                Shows all items of the sender's (or <user>'s if specified) list of items
+
+                who should <verb>(?)
+                Randomly selects a user from the orders list
+
+                help
+                Shows help (this section)
+            '''
+
+        return "I don't think I can do that. :frowning:"
+
+# User finder
+# Finds if a user exists
+# error is True, return an error message
+# error is False, create the user line
+def FIND_USER(user, error):
+    try:
+        ORDERS[user]
+    except KeyError:
+        if error:
+            return 'Sry, @**' + user + "** isn't hungry. :disappointed:"
+        else:
+            ORDERS[user] = []
+            return
+
+# Item interpreter
+# Transforms an array of items into a string
+def ITEM_INTERPRETER(items):
+    item_str = ''
+    for item in items:
+        if len(item_str) == 0:
+            item_str += ' ' + item
+        else:
+            if item_str[-1] == ',':
+                item_str += item
+            else:
+                item_str += ' ' + item
+
+    return item_str[1:].split(',')
+
+# Copies user list
+def copy_user_list(original_user, user):
+    # Verification
+    if FIND_USER(original_user, True):
```

Comment:
We should change the name of `FIND_USER` to be just `find_user` (lowercase).  More importantly, we want to figure out a nicer protocol for error handling, where you don't have to call back to the same function to get the error message.  We can have `find_user` return a tuple like `(True, ['ice cream'])` or `(False, 'Could not find Alice').  Or you may want to consider having a separate helper function to format error messages.